core/include/rng_support.h

```diff
@@ -7,6 +7,22 @@
 
 #include <stdint.h>
 
-TEE_Result hw_get_random_bytes(void *buf, size_t blen);
+/*
+ * hw_get_max_available_entropy() - Get the maximum bytes of entropy per call
+ * @blen:  [out] Maximum number of bytes of entropy that can be returned
+ *               in a single call to the hw_get_available_entropy() function
+ *
+ * Returns TEE_SUCCESS on success.
+ */
+TEE_Result hw_get_max_available_entropy(size_t *blen);
```

Comment:
+1, I find the similarity between `hw_get_max_available_entropy()` and `hw_get_available_entropy()` slightly confusing because one returns a number of bytes (entropy size) and the other returns entropy.